edx_rest_framework_extensions/authentication.py

```diff
@@ -0,0 +1,81 @@
+""" Authentication classes. """
+
+import logging
+
+import requests
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from rest_framework import exceptions
+from rest_framework.authentication import get_authorization_header, BaseAuthentication
+from rest_framework.status import HTTP_200_OK
+
+logger = logging.getLogger(__name__)
+User = get_user_model()
+
+
+class BearerAuthentication(BaseAuthentication):
+    """
+    Simple token based authentication.
+
+    Clients should authenticate by passing the token key in the "Authorization"
+    HTTP header, prepended with the string "Bearer ".  For example:
+
+        Authorization: Bearer 401f7ac837da42b97f613d789819ff93537bee6a
+    """
+
+    def get_provider_url(self):
+        """ Returns the root URL of the OAuth2 provider. """
+        return getattr(settings, 'OAUTH2_PROVIDER_URL', None)
+
+    def authenticate(self, request):
+        provider_url = self.get_provider_url()
+        if not provider_url:
+            return None
+
+        provider_url = provider_url.strip('/')
+
+        auth = get_authorization_header(request).split()
```

Comment:
As someone less familiar with the expected structure of this header string can we include a comment about what we are expecting to get back here.

It's not clear what an auth list contains, which is exacerbated further as this method continues to just index into the list to get what it needs.